packages/job-dispatch/src/policies/release-dependency.ts

```diff
@@ -1,138 +1,145 @@
-    );
-
-  return Promise.all(
-    jcs.map(async (jc) => {
-      if (jc.releaseDependencies.length === 0 || jc.target == null)
-        return jc.releaseJobTrigger;
-
-      const { targetMetadata } = jc;
-
-      const numDepsPassing = await Promise.all(
-        jc.releaseDependencies.map(async (rd) => {
-          const { releaseDependency: releaseDep, targetMetadataGroup: tlg } =
-            rd;
-
-          const relevantTargetMetadata = targetMetadata.filter((tm) =>
-            tlg.keys.includes(tm.key),
-          );
+  const passingReleasesJobTriggersPromises = releaseJobTriggers.map(
+    async (trigger) => {
+      const release = await db
+        .select()
+        .from(schema.release)
+        .innerJoin(
+          schema.releaseDependency,
+          eq(schema.release.id, schema.releaseDependency.releaseId),
+        )
+        .where(eq(schema.release.id, trigger.releaseId));
+
+      if (release.length === 0) return trigger;
+
+      const deps = release.map((r) => r.release_dependency);
+
+      if (deps.length === 0) return trigger;
+
+      const results = await db.execute(
+        sql`
+          WITH RECURSIVE reachable_relationships(id, visited, tr_id, source_id, target_id, type) AS (
+            -- Base case: start with the given ID and no relationship
+            SELECT 
+                ${trigger.targetId}::uuid AS id, 
+                ARRAY[${trigger.targetId}::uuid] AS visited,
+                NULL::uuid AS tr_id,
+                NULL::uuid AS source_id,
+                NULL::uuid AS target_id,
+                NULL::target_relationship_type AS type
+            UNION ALL
+            -- Recursive case: find all relationships connected to the current set of IDs
+            SELECT
+                CASE
+                    WHEN tr.source_id = rr.id THEN tr.target_id
+                    ELSE tr.source_id
+                END AS id,
+                rr.visited || CASE
+                    WHEN tr.source_id = rr.id THEN tr.target_id
+                    ELSE tr.source_id
+                END,
+                tr.id AS tr_id,
+                tr.source_id,
+                tr.target_id,
+                tr.type
+            FROM reachable_relationships rr
+            JOIN target_relationship tr ON tr.source_id = rr.id OR tr.target_id = rr.id
+            WHERE
+                NOT CASE
+                    WHEN tr.source_id = rr.id THEN tr.target_id
+                    ELSE tr.source_id
+                END = ANY(rr.visited)                
+                AND tr.target_id != ${trigger.targetId}
+        )
+        SELECT DISTINCT tr_id AS id, source_id, target_id, type
+        FROM reachable_relationships
+        WHERE tr_id IS NOT NULL;
+        `,
+      );
+
+      const relationships = results.rows.map((r) => ({
+        id: String(r.id),
+        sourceId: String(r.source_id),
+        targetId: String(r.target_id),
+        type: r.type as "associated_with" | "depends_on",
+      }));
```

Comment:
fix typing

Reply:
even if the sql operator is annotated with type, since its called in a db.execute the typing doesn't show up in the result. adding a comment explaining this wherever I do this